queries.sql

```diff
@@ -21,5 +21,83 @@ SELECT * FROM animals WHERE neutered = true;
 -- Find all animals not named Gabumon.
 SELECT * FROM animals WHERE NOT name = 'Gabumon';
 
--- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
+/* 
+** Find all animals with a weight between 10.4kg and 17.3kg (including the 
+** animals with the weights that equals precisely 10.4kg or 17.3kg)
+*/
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+-- Start a transaction to set all species to 'unspecified' and roll it back afterwards
+BEGIN;
+UPDATE animals SET species='unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+
+
+-- Start a transaction and persist the data afterwards
+BEGIN:
+/*
+Update the animals table by setting the species column
+** to digimon for all animals that have a name ending in mon
+*/
+UPDATE animals SET species='pigimon' WHERE name LIKE '%mon';
+/*
+** Update the animals table by setting the species column
+** to pokemon for all animals that don't have species already set.
+*/
+UPDATE animals SET species='pokemon' WHERE name NOT LIKE '%mon';
+-- Commit the transaction.
+COMMIT;
+-- Verify that change was made and persists after commit.
+SELECT * FROM animals;
+
+
+-- Inside a transaction delete all records and roll it back.
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+
+-- Begin a transaction.
+BEGIN;
+-- Delete all animals born after Jan 1st, 2022.
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+-- Create a savepoint for the transaction.
+SAVEPOINT deleted_records_after_jan_01_2022;
+-- Update all animals' weight to be their weight multiplied by -1.
+UPDATE animals SET weight_kg = weight_kg * -1;
+-- Rollback to the savepoint.
+ROLLBACK TO deleted_records_after_jan_01_2022;
+-- Update all animals' weights that are negative to be their weight multiplied by -1.
+UPDATE animals 
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+-- Commit transaction
+COMMIT;
+
+
+-- How many animals are there?
+SELECT count(*) from animals;
+-- How many animals have never tried to escape?
+SELECT count(*) from animals WHERE escape_attempts = 0;
+-- What is the average weight of animals?
+SELECT SUM(weight_kg) / count(*) AS average_weight FROM animals;
+-- Takes care of null weight if any
+
+-- Who escapes the most, neutered or not neutered animals?
+select neutered, sum from (
+  select neutered, sum(escape_attempts) from animals
+  group by neutered) as foo
+  where sum = (
+    select max(sum) from (
+      select neutered, sum(escape_attempts) from animals
+      group by neutered
+  ) as bar
+);
```

Comment:
[Optional]
- [ ] Again SQL's AVG function can help you in these situations.
```suggestion
   SELECT neutered, AVG(escape_attempts) FROM animals
   GROUP BY neutered;
```

Reply:
The question I want to answer here is:
Who escapes the most, neutered or not neutered animals?
I think that the answer to this question is finding which animals escape the most. Is it the neutered or not neutered animals? I summed the escape attempts for the neutered and also for the not neutered. Finally, I checked which between them had the maximum number of escape attempts. I think I needed to return either neutered or not neutered and that is what I did.
Please, check this:point_down:
![image](https://user-images.githubusercontent.com/45668340/138975017-01477091-916b-426e-8bc7-723eaacfc9f5.png)
If you check the total escape attempts for all the neutered animals it gives 18 escape attempts, which is more than that of the not neutered (4 escape attempts).
What do you think?:thinking:

Reply:
What you're doing is right, and it might be better if you're doing data analysis.  I just wanted you to know that you can find it easily as I suggested.  That's why I marked it as optional.  The group that escapes the most will always have the higher average escape.

Reply:
I think you are right but I also imagined another scenario where specie A has 3 records with 20 escape attempts each. Then, specie B has 4 records with three of them having same escape attempts as that of A (20 each). 

If the fourth has escape attempts of 2 for example, the total escape attempts for specie B would be greater but it's average will, at the same time, be less than that of A. Please, does it make sense?

Though I had submitted another pull request for a re-review before I received your last comment, I would consider using your approach still later.